server/bleep/src/cache.rs

```diff
@@ -149,8 +163,102 @@ impl<'a> FileCache<'a> {
         Ok(())
     }
 
-    pub async fn chunks_for_file(&self, key: &'a str) -> ChunkCache<'a> {
-        ChunkCache::for_file(self.db, self.reporef, key).await
+    /// Commit the embed log, invoking the embedder if batch size is met.
+    ///
+    /// If `flush == true`, drain the log, send the entire batch to
+    /// the embedder, and commit the results, disregarding the internal
+    /// batch sizing.
+    pub async fn commit_embed_queue(&self, flush: bool) -> anyhow::Result<()> {
+        let Some(semantic) = self.semantic
+	else {
+	    return Ok(());
+	};
+
+        let new_points = self.embed_queued_points(semantic, flush).await?;
+
+        if !new_points.is_empty() {
+            if let Err(err) = semantic
+                .qdrant_client()
+                .upsert_points(&semantic.collection_name(), new_points, None)
+                .await
+            {
+                error!(?err, "failed to write new points into qdrant");
+            }
+        }
+        Ok(())
+    }
+
+    async fn embed_queued_points(
```

Comment:
```suggestion
    async fn try_embed_queued_points(
```
Can we rename this, to make it clearer that this method will often do nothing?